### PR TITLE
Add support for multiple releases in documentation

### LINKED ISSUE #8: Wrong handling of versions in documentation
### Bug Report

Different versions can be handled by our documentation theme.

laminas-hydrator use different versions in documentation already:
https://github.com/laminas/laminas-hydrator/blob/02a5cfdc0dbf2b9abbe7aca3472de514d10e2d6b/mkdocs.yml#L6-L27

### PR DESCRIPTION
**Pull request by tux-rampage**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | yes
| BC Break      | no
| New Feature   | yes
| RFC           | yes
| QA            | yes

### Description

This PR aims to add support for multiple versions/releases to the documentation pages as requested in #8

### CODE REVIEW

**Comment by tux-rampage** [comment]
@froschdesign Should we move the current docs to a `v3` subdirectory? I guess this will require to modify all page references to be prefixed with `v3/` then, right?

File: mkdocs.yml [hunk 1]
```diff
@@ -2,29 +2,37 @@ docs_dir: docs/book
...
+            - Configuration: config.md
+            - "Injector": injector.md
+            - "Code Generator": codegen.md
+        - Cookbook:
+            - "Usage with PSR-11 Containers": cookbook/use-with-psr-containers.md
+            - "Usage with laminas-servicemanager": cookbook/use-with-servicemanager.md
+            - "Using AoT with Mezzio and laminas-servicemanager": cookbook/aot-guide.md
+        - "Migration Guide": migration.md
+    - v2:
+        - "Quick Start": v2/quick-start.md
```
Comment (tux-rampage) [inline mkdocs.yml:20]: I restored this page from the 2.x releases.

File: mkdocs.yml [hunk 2]
```diff
@@ -2,29 +2,37 @@ docs_dir: docs/book
...
-        - "v2 Documentation":
+    - v3:
+        - Introduction: intro.md
+        - Installation: installation.md
+        - "Quick Start": quick-start.md
+        - Reference:
+            - "PSR-11 Support": psr-11.md
+            - Configuration: config.md
+            - "Injector": injector.md
+            - "Code Generator": codegen.md
```
Comment (froschdesign) [inline mkdocs.yml]:
```suggestion
        - "PSR-11 Support": psr-11.md
        - Configuration: config.md
        - "Injector": injector.md
        - "Code Generator": codegen.md
```

File: mkdocs.yml [hunk 3]
```diff
@@ -2,29 +2,37 @@ docs_dir: docs/book
...
+        - Cookbook:
+            - "Usage with PSR-11 Containers": cookbook/use-with-psr-containers.md
+            - "Usage with laminas-servicemanager": cookbook/use-with-servicemanager.md
+            - "Using AoT with Mezzio and laminas-servicemanager": cookbook/aot-guide.md
+        - "Migration Guide": migration.md
```
Comment (froschdesign) [inline mkdocs.yml]:
```suggestion
        - Migration:
            - "Migration from Version 2 to 3": migration.md
```

File: docs/book/v2/quick-start.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
+# Quick Start
+
+This quick start is intended to get developers familiar with the concepts of the
+laminas-di DiC.  Generally speaking, code is never as simple as it is inside this
+example, so working knowledge of the other sections of the manual is suggested.
+
+Assume, for a moment, the following application code. It already assumes that
```
Comment (froschdesign) [inline docs/book/v2/quick-start.md:9]: A headline can be added here. For example: "Example Application"

File: docs/book/v2/quick-start.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+If you are doing this above wiring in each controller or view that wants to list
+movies, not only can this become repetitive and boring to write, but also
+unmaintainable if you want to swap out one of these dependencies on a wholesale
+scale.
+
+Since this example of code already practices good dependency injection using
+constructor injection, it is a great candidate for using laminas-di.
+
+The following demonstrates how to wire the above into a laminas-di container:
```
Comment (froschdesign) [inline docs/book/v2/quick-start.md:76]: A headline can be added here. For example: "Using a laminas-di Container"

File: docs/book/v2/quick-start.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+This means that when `$di->get()` is called, it will be consulting the
+`RuntimeDefinition`, which uses Reflection to understand the structure of the
+code. Once it knows the structure of the code, it can then know how the
+dependencies fit together and how to go about wiring your objects for you.
+`Laminas\Di\Definition\RuntimeDefinition` will utilize the names of the parameters
+in the methods as the class parameter names. This is how both the `username` and
+`password` keys are mapped to the first and second parameters, respectively, of
+the constructor consuming these named parameters.
+
+If you were to want to pass in the username and password at call time, this is
```
Comment (froschdesign) [inline docs/book/v2/quick-start.md:118]: A headline can be added here. For example: "Passing Parameters on Call Time"

**Review by froschdesign — CHANGES_REQUESTED** [review]
Only some small changes and improvements are needed.

File: docs/book/v2/quick-start.md (hunk 5 above)
Comment (tux-rampage) [reply docs/book/v2/quick-start.md:76]: Oops, added the headline a paragraph too early. Should be fixed now.

File: docs/book/v2/quick-start.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+instance manager and no configuration:
+
+```php
+public function __construct(
+    DefinitionList $definitions = null,
+    InstanceManager $instanceManager = null,
+    Configuration $config = null
+) {
+    $this->definitions = ($definitions) ?: new DefinitionList(new Definition\RuntimeDefinition());
+    $this->instanceManager = ($instanceManager) ?: new InstanceManager();
```
Comment (froschdesign) [inline docs/book/v2/quick-start.md]: I think the full class names with namespaces should be here.

Comment (tux-rampage) [reply docs/book/v2/quick-start.md]: This is (incorrectly) mirroring to the constructor code of `Laminas\Di\Di` (v2): https://github.com/laminas/laminas-di/blob/2.6.1/src/Di.php#L113

Comment (tux-rampage) [reply docs/book/v2/quick-start.md]: I changed it to an example code snippet

**Review by froschdesign — APPROVED** [review]

**Pull request merged**